### PR TITLE
NTP-544: back from search results to key stage and continue clears subjects

### PR DESCRIPTION
**Pull request by stephenjmcneill**
## Context

Data selections are lost when navigating back along the application's journey

## Guidance to review

Go back and forth along the Find a Tutor journey and make sure selections are maintained and edits work. 

## Link to Jira ticket

https://dfedigital.atlassian.net/browse/NTP-544

## Things to check

- [x] Title is in the format "NTP-XXX: Short description" where NTP-XXX is the Jira ticket reference
- [x] Code and tests follow the [coding standards](/docs/coding-standards.md)
- [x] `dotnet format` has been run in the repository root
- [ ] Test coverage of new code is at least 80%
- [ ] All UI related acceptance criteria specified in the ticket have been added to the relevant [cypress test feature file(s)](/UI/cypress/e2e/)
- [ ] Unless explicitly mentioned in the ticket, all UI work should have been tested as working on desktop and mobile resolutions with JavaScript on and off
- [ ] Database migrations can be applied to the current codebase in main without causing exceptions
- [ ] Debug logging has been added after all logic decision points
- [ ] All [automated testing](/README.md#testing) including accessibility and security has been run against your code
- [ ] **PR deployment has been signed off by wider team**

### CODE REVIEW

File: UI/cypress/e2e/tuition-partner-user-journey.feature [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Feature: User can travel forwards and backwards
+
+Scenario: User can move forward in the application to the end of the journey
+   Given a user has started the 'Find a tuition partner' journey
+    Then they will be able journey forward to a selected tuition partner page
+
+Scenario: User can travel back to the begining of journey
+    Given a user has started the 'Find a tuition partner' journey
+        And they will be able journey forward to a selected tuition partner page
+    Then they will be journey back to the page they started from
```
Comment (benagain) [inline UI/cypress/e2e/tuition-partner-user-journey.feature:10]: These sentences seem a little disjointed - "And they will be able journey...".

**Review by benagain — APPROVED** [review]
Great comprehensive testing

File: UI/Pages/Index.cshtml [hunk 2]
```diff
@@ -34,6 +34,25 @@
...
+				}
+			}
+			@if (Model.Data.Subjects != null)
+			{
+			    foreach (var subject in Model.Data.Subjects)
+			    {
+					<input type="hidden" id="@subject" name="Data.Subjects" value="@subject" />
+				}
+			}
+            @if (Model.Data.TuitionType.HasValue)
```
Comment (benagain) [inline UI/Pages/Index.cshtml:51]: I'm not sure that this check is necessary, The hidden fields will simply have no value if the property is null.

File: UI/cypress/e2e/tuition-partner-user-journey.feature (hunk 1 above)
Comment (stephenjmcneill) [reply UI/cypress/e2e/tuition-partner-user-journey.feature:10]: Changed

**Review by benagain — APPROVED** [review]

**Pull request merged**